src/osemosys2iamc/resultify.py

```diff
@@ -318,95 +281,145 @@ def main(config: Dict, inputs_path: str, results_path: str) -> pyam.IamDataFrame
+                    data = filter_capacity(results, technologies)
+                elif 'excluded_prod_tech' in result.keys():
+                    technologies = result['excluded_prod_tech']
+                    data = filter_capacity(results, technologies)
+                elif 'el_prod_technology' in result.keys():
+                    technologies = result['el_prod_technology']
+                    data = filter_capacity(results, technologies)
+                elif 'demand' in result.keys():
+                    demands = result['demand']
+                    data = filter_final_energy(results, demands)
+                else:
+                    data = extract_results(results, technologies)
 
-        else:
-            results = {}
-            for p in result['osemosys_param']:
-                inpathname = os.path.join(results_path, p + '.csv')
-                results[p] = read_file(inpathname)
-            if 'trade_tech' in result.keys():
-                technologies = result['trade_tech']
-                data = calculate_trade(results, technologies)
+            else:
+                results = {}
+                for p in result['osemosys_param']:
+                    path_name = os.path.join(results_path, p + '.csv')
+                    results[p] = read_file(path_name)
+                if 'trade_tech' in result.keys():
+                    technologies = result['trade_tech']
+                    data = calculate_trade(results, technologies)
+
+            if 'transform' in result.keys():
+                if result['transform'] == 'abs':
+                    data['VALUE'] = data['VALUE'].abs()
+
+            if not data.empty:
+                data = data.rename(columns={'REGION': 'region',
+                                            'YEAR': 'year',
+                                            'VALUE': 'value'})
+                print(data)
+                iamc = pyam.IamDataFrame(
+                    data,
+                    model=config['model'],
+                    scenario=config['scenario'],
+                    variable=result['iamc_variable'],
+                    unit=unit)
+                blob.append(iamc)
+    except KeyError:
+        pass
 
-        aggregated = aggregate(data)
+    all_data = pyam.concat(blob)
 
-        if not aggregated.empty:
-            iamc = make_iamc(aggregated, config['model'], config['scenario'], result['iamc_variable'], unit)
-            blob.append(iamc)
+    all_data = all_data.convert_unit('PJ/yr', to='EJ/yr')
+    all_data = all_data.convert_unit('ktCO2/yr', to='Mt CO2/yr', factor=0.001)
+    all_data = all_data.convert_unit('MEUR_2015/PJ', to='EUR_2020/GJ', factor=1.05)
+    all_data = all_data.convert_unit('kt CO2/yr', to='Mt CO2/yr')
 
-    all_data = pyam.concat(blob)
+    if iso_mapping is not None:
+        data = all_data.timeseries()
+        data.index = data.index.set_levels(data.index.levels[2].map(iso_mapping), level=2)
```

Comment:
The problem described below originates here. It seems the `iso_mapping` is not converting `NO` to `Norway`, but to `nan`. Any idea what might be the reason for this @danielhuppmann ? 
i checked the definition in the openentrance package and `NO` is the iso2 code for Norway.